ban-cooking.lua

```diff
@@ -49,7 +27,7 @@ local function reaction_product_id_contains(reaction_product, str)
 end
 
 local function ban_cooking(print_name, mat_type, mat_index, type, subtype)
-    key = mat_type, mat_index, type, subtype
+    local key = mat_type
```

Comment:
this confuses me. I said that the previous statement was equivalent to this one, but the implication was that this was wrong. Keys should encompass everything you need to differentiate the tuple, but keys cannot be tables. I'm pretty sure you should be stringifying the four tuple of elements. if you still need to address them individually, assign the table of separate elements as the value instead of `true`. a table is still "truthy" so it can still be easily tested for membership.